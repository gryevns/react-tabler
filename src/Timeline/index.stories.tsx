import React from "react";
import { Briefcase, Check, Image, Twitter, UserPlus } from "react-feather";

import Avatar from "../Avatar";
import Card from "../Card";
import Timeline from "../Timeline";
import { boolean, withKnobs } from "@storybook/addon-knobs";

export default {
    title: "Timeline",
    decorators: [withKnobs],
};

export const History = () => (
    <Card>
        <Card.Body>
            <Timeline simple={boolean("Simple", false)}>
                <Timeline.Item>
                    <Timeline.Icon className="bg-twitter">
                        <Twitter size={16} />
                    </Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>10 hrs ago</Timeline.Time>
                        <Timeline.Title>+1150 Followers</Timeline.Title>
                        <p className="text-muted">
                            Youâ€™re getting more and more followers, keep it up!
                        </p>
                    </Timeline.Content>
                </Timeline.Item>
                <Timeline.Item>
                    <Timeline.Icon className="bg-red">
                        <Briefcase size={16} />
                    </Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>2 hrs ago</Timeline.Time>
                        <Timeline.Title>+3 New Products were added!</Timeline.Title>
                        <p className="text-muted">Congratulations!</p>
                    </Timeline.Content>
                </Timeline.Item>
                <Timeline.Item>
                    <Timeline.Icon className="bg-success">
                        <Check size={16} />
                    </Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>1 day ago</Timeline.Time>
                        <Timeline.Title>Database backup completed!</Timeline.Title>
                        <p className="text-muted">
                            Download the <a href="#">latest backup</a>.
                        </p>
                    </Timeline.Content>
                </Timeline.Item>
                <Timeline.Item>
                    <Timeline.Icon className="bg-teal">
                        <UserPlus size={16} />
                    </Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>2 days ago</Timeline.Time>
                        <Timeline.Title>+3 Friend Requests</Timeline.Title>
                        <Avatar.List className="mt-3">
                            <Avatar status="green" />
                            <Avatar status="green">JL</Avatar>
                            <Avatar status="green" />
                        </Avatar.List>
                    </Timeline.Content>
                </Timeline.Item>
                <Timeline.Item>
                    <Timeline.Icon className="bg-yellow">
                        <Image size={16} />
                    </Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>3 days ago</Timeline.Time>
                        <Timeline.Title>+2 New photos</Timeline.Title>
                        <div className="mt-3">
                            <img
                                src="https://picsum.photos/id/128/128"
                                className="mr-3"
                            />
                            <img src="https://picsum.photos/id/127/128" />
                        </div>
                    </Timeline.Content>
                </Timeline.Item>
            </Timeline>
        </Card.Body>
    </Card>
);
