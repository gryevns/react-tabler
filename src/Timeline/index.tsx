import React from "react";
import cn from "classnames";

export interface TimelineTitleProps {
    children: React.ReactNode;
}

const TimelineTitle = ({ children }: TimelineTitleProps) => (
    <p className="list-timeline-title">{children}</p>
);

export interface TimelineContentProps {
    children?: React.ReactNode;
}

const TimelineContent = ({ children }: TimelineContentProps) => (
    <div className="list-timeline-content">{children}</div>
);

export interface TimelineTimeProps {
    children?: React.ReactNode;
}

const TimelineTime = ({ children }: TimelineTimeProps) => (
    <div className="list-timeline-time">{children}</div>
);

export interface TimelineIconProps {
    children: React.ReactNode;
    className?: string;
}

const TimelineIcon = ({ className, children }: TimelineIconProps) => {
    const classes = cn("list-timeline-icon", className);
    return <div className={classes}>{children}</div>;
};

export interface TimelineItemProps {
    children: React.ReactNode;
}

const TimelineItem = ({ children }: TimelineItemProps) => <li>{children}</li>;

export interface TimelineProps {
    children: React.ReactNode;
    className?: string;
    simple?: boolean;
}

const Timeline = ({ children, className, simple = false }: TimelineProps) => {
    const classes = cn(
        "list list-timeline",
        { "list-timeline-simple": simple },
        className,
    );
    return <ul className={classes}>{children}</ul>;
};

Timeline.Item = TimelineItem;
Timeline.Icon = TimelineIcon;
Timeline.Time = TimelineTime;
Timeline.Title = TimelineTitle;
Timeline.Content = TimelineContent;

export default Timeline;
