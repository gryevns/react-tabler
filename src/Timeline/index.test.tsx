import React from "react";
import { shallow } from "enzyme";

import Timeline from "./index";

describe("Timeline", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
            <Timeline>
                <Timeline.Item>
                    <Timeline.Icon className="bg-twitter">icon</Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>10 hrs ago</Timeline.Time>
                        <Timeline.Title>+1150 Followers</Timeline.Title>
                        You’re getting more and more followers, keep it up!
                    </Timeline.Content>
                </Timeline.Item>
            </Timeline>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match simple snapshot", () => {
        const wrapper = shallow(
            <Timeline simple>
                <Timeline.Item>
                    <Timeline.Icon className="bg-twitter">icon</Timeline.Icon>
                    <Timeline.Content>
                        <Timeline.Time>10 hrs ago</Timeline.Time>
                        <Timeline.Title>+1150 Followers</Timeline.Title>
                        You’re getting more and more followers, keep it up!
                    </Timeline.Content>
                </Timeline.Item>
            </Timeline>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
