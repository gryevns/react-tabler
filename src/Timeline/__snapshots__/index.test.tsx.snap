// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should match simple snapshot 1`] = `
<ul
  className="list list-timeline list-timeline-simple"
>
  <TimelineItem>
    <TimelineIcon
      className="bg-twitter"
    >
      icon
    </TimelineIcon>
    <TimelineContent>
      <TimelineTime>
        10 hrs ago
      </TimelineTime>
      <TimelineTitle>
        +1150 Followers
      </TimelineTitle>
      You’re getting more and more followers, keep it up!
    </TimelineContent>
  </TimelineItem>
</ul>
`;

exports[`Timeline should match snapshot 1`] = `
<ul
  className="list list-timeline"
>
  <TimelineItem>
    <TimelineIcon
      className="bg-twitter"
    >
      icon
    </TimelineIcon>
    <TimelineContent>
      <TimelineTime>
        10 hrs ago
      </TimelineTime>
      <TimelineTitle>
        +1150 Followers
      </TimelineTitle>
      You’re getting more and more followers, keep it up!
    </TimelineContent>
  </TimelineItem>
</ul>
`;
